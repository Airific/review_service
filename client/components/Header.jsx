/* eslint-disable react/prop-types */
import React from 'react';
import Styled from 'styled-components';

const Star = Styled.span`
  font-size: ${(props) => (props.isOpen ? '27px' : '21px')};
  color: rgb(255, 56, 92);
  width: 20px;
  margin-right: ${(props) => (props.isOpen ? '10px' : '0px')};
`;

const Grid = Styled.span`
  display: grid;
  grid-template-columns: 1.5fr auto 6fr;
  width: 300px;
  align-items: center;
  line-height: 30px;
`;

const Average = Styled.div`
  font-weight: 500;
  padding-left: 3px;
  font-size: ${(props) => (props.isOpen ? '32px' : '21px')};
  color: #222222;
  background-color: #fff;
  letter-spacing: normal;
`;

const Header = ({ rating, reviews, isOpen }) => (
  <Grid>
    <Star isOpen={isOpen}>&#9733;</Star>
    <Average isOpen={isOpen}>
      {`${rating.total} (${reviews.length} reviews)`}
    </Average>
  </Grid>
);

export default Header;
