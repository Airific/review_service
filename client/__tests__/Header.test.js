/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

const reviews = [
  {
    name: 'test',
    date: 'May',
    photo: 'https://airific-review-images.s3.us-east-2.amazonaws.com/image20.jpg',
    text: 'Hello!',
  },
  {
    name: 'alissa',
    date: 'June',
    photo: 'https://airific-review-images.s3.us-east-2.amazonaws.com/image20.jpg',
    text: 'test!',
  },
  {
    name: 'hack reactor',
    date: 'October',
    photo: 'https://airific-review-images.s3.us-east-2.amazonaws.com/image20.jpg',
    text: 'Learn to code!',
  },
];
const rating = {
  communication: '5',
  location: '3',
  clean: '2',
  accuracy: '1',
  value: '4',
  check: '3',
  total: '3.45',
};
describe('Header component', () => {
  it('render correctly Header component', () => {
    const wrapper = shallow(<Header reviews={reviews} rating={rating} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the correct amount of reviews and correct rating', () => {
    const wrapper = mount(<Header reviews={reviews} rating={rating} />);
    expect(wrapper.find('div').text()).toEqual('3.45 (3 reviews)');
  });
});
