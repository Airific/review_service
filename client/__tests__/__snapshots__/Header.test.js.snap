// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render correctly Header component 1`] = `
<styled.span>
  <styled.span>
    â˜…
  </styled.span>
  <styled.div>
    3.45 (3 reviews)
  </styled.div>
</styled.span>
`;
